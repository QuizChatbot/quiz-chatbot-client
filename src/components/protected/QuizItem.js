import React, { Component } from 'react'
import PropTypes from 'prop-types'
import QuizInput from './QuizInput'
import styled from 'styled-components'
import IconButton from 'material-ui/IconButton'
import ActionDelete from 'material-ui/svg-icons/action/delete'
import Paper from 'material-ui/Paper'

const PaperStyled = styled(Paper)`
  text-align: left;
`

export const Element = ({ idx, quest, deleteQuiz, editQuiz }) => {
  function handleSave (id, quiz, isChoice) {
    editQuiz(`${id}`, quiz, isChoice)
  }

  return (
    <div>
      <QuizInput
        idx={idx}
        quest={quest}
        onSave={(quiz, isChoice) => handleSave(quest.id, quiz, isChoice)}
      />
      <div style={{ textAlign: 'right' }}>
        <IconButton tooltip='Delete'>
          <ActionDelete onTouchTap={() => deleteQuiz(quest.id)} />
        </IconButton>
      </div>
    </div>
  )
}

class QuizItem extends Component {
  render () {
    const { idx, quest, deleteQuiz, editQuiz } = this.props
    return (
      <PaperStyled zDepth={2}>
        <Element
          idx={idx}
          quest={quest}
          deleteQuiz={deleteQuiz}
          editQuiz={editQuiz}
        />
      </PaperStyled>
    )
  }
}

QuizItem.propTypes = {
  idx: PropTypes.number,
  quest: PropTypes.object,
  editQuiz: PropTypes.func,
  deleteQuiz: PropTypes.func
}

export default QuizItem
