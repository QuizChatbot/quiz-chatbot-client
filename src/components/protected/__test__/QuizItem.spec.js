import React from 'react'
import { shallow, mount, render } from 'enzyme'
import QuizItem, { Element } from '../QuizItem'
import IconButton from 'material-ui/IconButton'

describe('QuizItem component', () => {
  const quest = {}
  const func = () => {}

  it('should render Element component', () => {
    const Wrapper = shallow(
      <QuizItem idx={0} quest={quest} editQuiz={func} deleteQuiz={func} />
    )
    expect(
      Wrapper.contains(
        <Element idx={0} quest={quest} editQuiz={func} deleteQuiz={func} />
      )
    ).toEqual(true)
  })

  it('Element component should render QuizInput component and delete button', () => {
    const Wrapper = shallow(
      <Element idx={0} quest={quest} deleteQuiz={func} editQuiz={func} />
    )
    expect(Wrapper.find('QuizInput')).toHaveLength(1)
    expect(Wrapper.find('IconButton')).toHaveLength(1)
  })
})
