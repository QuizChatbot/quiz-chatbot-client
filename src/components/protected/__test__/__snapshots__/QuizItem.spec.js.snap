// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizItem component should render Element component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Styled(Paper)
    zDepth={2}
>
    <Unknown
        deleteQuiz={[Function]}
        editQuiz={[Function]}
        handleClose={[Function]}
        handleOpen={[Function]}
        idx={0}
        isDialogOpen={false}
        quest={Object {}}
    />
</Styled(Paper)>,
  "nodes": Array [
    <Styled(Paper)
      zDepth={2}
>
      <Unknown
            deleteQuiz={[Function]}
            editQuiz={[Function]}
            handleClose={[Function]}
            handleOpen={[Function]}
            idx={0}
            isDialogOpen={false}
            quest={Object {}}
      />
</Styled(Paper)>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <QuizItem
        deleteQuiz={[Function]}
        editQuiz={[Function]}
        idx={0}
        quest={Object {}}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": QuizItem {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleClose": [Function],
        "handleOpen": [Function],
        "props": Object {
          "deleteQuiz": [Function],
          "editQuiz": [Function],
          "idx": 0,
          "quest": Object {},
        },
        "refs": Object {},
        "state": Object {
          "isDialogOpen": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Styled(Paper)
          zDepth={2}
>
          <Unknown
                    deleteQuiz={[Function]}
                    editQuiz={[Function]}
                    handleClose={[Function]}
                    handleOpen={[Function]}
                    idx={0}
                    isDialogOpen={false}
                    quest={Object {}}
          />
</Styled(Paper)>,
        "_debugID": 2,
        "_renderedOutput": <Styled(Paper)
          zDepth={2}
>
          <Unknown
                    deleteQuiz={[Function]}
                    editQuiz={[Function]}
                    handleClose={[Function]}
                    handleOpen={[Function]}
                    idx={0}
                    isDialogOpen={false}
                    quest={Object {}}
          />
</Styled(Paper)>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <QuizItem
    deleteQuiz={[Function]}
    editQuiz={[Function]}
    idx={0}
    quest={Object {}}
/>,
}
`;
